clusters/app.ci/supplemental-ci-images/dynamic-plugin-sdk-buildroot.yaml

```diff
@@ -0,0 +1,61 @@
+apiVersion: image.openshift.io/v1
+kind: ImageStream
+metadata:
+  name: openshift-dynamic-plugin-sdk-buildroot
+  namespace: ci
+---
+apiVersion: build.openshift.io/v1
+kind: BuildConfig
+metadata:
+  name: openshift-dynamic-plugin-sdk-buildroot
+  namespace: ci
+spec:
+  successfulBuildsHistoryLimit: 1
+  failedBuildsHistoryLimit: 2
+  output:
+    to:
+      kind: ImageStreamTag
+      name: openshift-dynamic-plugin-sdk-buildroot:nodejs16
+      namespace: ci
+  source:
+    dockerfile: |
+      FROM quay.io/centos/centos:stream8
+
+      ENV WORKDIR=/go
+
+      RUN yum install -y epel-release && \
+          rpm -V epel-release && \
+          INSTALL_PKGS="bc bind-utils bsdtar device-mapper device-mapper-persistent-data e2fsprogs ethtool file findutils gcc git glib2-devel hostname iptables jq krb5-devel libseccomp-devel libvirt-devel lsof make mercurial nmap-ncat openssl rsync socat systemd-devel tar tito tree util-linux wget which xfsprogs zip" && \
+          yum install -y $INSTALL_PKGS && \
+          rpm -V $INSTALL_PKGS && \
+          yum clean all && \
+          touch /os-build-image && \
+          git config --system user.name origin-release-container && \
+          git config --system user.email origin-release@redhat.com && \
+          mkdir $WORKDIR
+
+      ENV VAULT_DISABLE_MLOCK true
+      RUN yum-config-manager --add-repo https://rpm.releases.hashicorp.com/RHEL/hashicorp.repo && \
```

Comment:
@bear-redhat can you chime in if there is documentation around which of these packages is required? @vojtechszocs and I noticed `mercurial` is installed above and we likely don't need it for example.

Reply:
We could reuse this docker image https://github.com/RedHatInsights/insights-frontend-builder-common/blob/master/src/releaser.Dockerfile We could add some additional dependencies if needed.

Reply:
@vojtechszocs what is your opinion here? I'm fine with either approach TBH, though I agree we should converge on a single "nodejs-16" image for our projects.

Reply:
The Dockerfile @karelhala referred to includes some things we don't need, such as `golang` package and `github.com` ssh host config.

I'd suggest to go with a minimal Dockerfile that provides
- explicit control over which Node.js version is installed (don't rely on what is provided by default)
- meets the minimum filesystem & package requirements of OpenShift `ci-operator` [1]
  - installed `git` package
  - writable `/go` directory
- npm package manager updated to latest stable version
- Yarn installed as a global npm package

[1] https://docs.ci.openshift.org/docs/architecture/ci-operator/#build-root-image

Reply:
```Dockerfile
FROM quay.io/fedora/fedora:36-x86_64

RUN dnf update -y && \
    dnf install git jq -y && \
    git config --system user.name origin-release-container && \
    git config --system user.email origin-release@redhat.com && \
    dnf module enable nodejs:16 -y && \
    dnf module install nodejs:16 -y && \
    WORKDIR='/go' && \
    mkdir $WORKDIR && \
    npm config -g set cache $WORKDIR/.npm && \
    npm install -g --no-update-notifier npm && \
    npm install -g --no-update-notifier yarn && \
    dnf clean all
```

Reply:
I've tested the above Dockerfile - it uses Node.js `v16.15.0` and npm `8.19.2`

```
$ docker images 
REPOSITORY              TAG         IMAGE ID       CREATED          SIZE
test                    latest      1da3243d7710   14 minutes ago   424MB
quay.io/fedora/fedora   36-x86_64   2ecb6df95994   8 weeks ago      163MB
```

The image size is relatively small (424MB) compared to the image based on CentOS 8 :smiley: 

The build log is here: [Test.dockerfile.log](https://github.com/openshift/release/files/9577951/Test.dockerfile.log)

Reply:
Done!